src/parallel/parallel.cpp

```diff
@@ -193,10 +359,39 @@ bool
 ParallelRobinHoodHashTable::locked_insert(ParallelBucket &entry_to_insert, size_t swap_index)
 {
     // TODO
+    //size_t swap_index = entry_to_insert.offset; //? i thnk its redoing it bc of possible contention but its locked so??
+    
+    ThreadManager manager = this->get_thread_lock_manager(); 
+    while(true) {
+        ParallelBucket &swapped_entry = this->do_atomic_swap(entry_to_insert, swap_index);
+        if(swapped_entry.offset == SIZE_MAX) { //size max is empty 
+            return true;
+        }
+        size_t swap_index_copy = swap_index;
+        auto [swap_index, is_found] = this->find_next_index_lock(manager, swap_index_copy, entry_to_insert.key, entry_to_insert.offset);
```

Comment:
```suggestion
        auto [next_swap_index, found] = this->find_next_index_lock(manager, swap_index, entry_to_swap.key, entry_to_swap.offset);
        swap_index = next_swap_index;
```
A bit easier to read this way.